analyses/bangladesh/trigger_development/bgd-glofas-new-model.md

```diff
@@ -0,0 +1,404 @@
+    df_ds = get_detection_stats(df_final, glofas_var_name,
+                                thresh_array)
+    i = np.argmin(np.abs(thresh_array - trigger_val))
+    print(f'Stats for trigger value of {trigger_val}:')
+    print(f'TP: {df_ds["TP"][i]:.0f}, FP: {df_ds["FP"][i]:.0f}, FN: {df_ds["FN"][i]:.0f}')
+    
+def plot_offset_days(df_final, glofas_var_name, thresh_array):
+    df_ds = get_detection_stats(df_final, glofas_var_name, thresh_array)
+    fig, ax = plt.subplots()
+    for i, row in df_ds.iterrows():
+        y = row['day_offset_reduced']
+        x = [row['thresh']] * len(y)
+        ax.plot(x, y, '.', c='C0', alpha=0.2)
+        if row['thresh'] == 97000:
+            ax.plot(x, y, '.', c='C1', alpha=0.4)
+    ax.axhline(0, c='k', ls=':', lw=0.5, alpha=0.5)
+    ax.set_ylabel('Detection day offset')
+    x_axis_units='[m$^3$ s$^{-1}$]'
+    ax.set_xlabel(f'GloFAS trigger threshold {x_axis_units}')
+```
+
+```python
+thresh_array = np.arange(70000, 110000, 500)
+glofas_var_name = 'glofas_observed'
+
+plot_stats(df_final, glofas_var_name, thresh_array)
+plot_offset_days(df_final, glofas_var_name, thresh_array)
+
+# Print out last year's trigger stats for 1 in 5 year val
+print_stats_for_val(df_final, glofas_var_name, thresh_array, rp_dict[5])
+
+
+```
+
+```python
+def plot_years(df_final, thresh, glofas_var='glofas_observed', glofas_xlims=(40000, 140000), forecast_var=None):
+    fig, axs = plt.subplots(17, 2, figsize=(10, 30))
+    fig.autofmt_xdate()
+    axs = axs.flat
+    df_final['year'] = df_final.index.year
+    iax = 0
+    for year, df in df_final.groupby('year'):
+        if year == 2021:
+            break
+        ax1 = axs[iax]
+        x = df.index
+        y1 = df['observed']
+        ax1.plot(x, y1, '.')
+        idx_event = df['event'] == True
+        ax1.plot(x[idx_event], y1[idx_event], 'or', ms=5)
+        ax1.set_xlim(datetime(year, 3, 1), datetime(year, 10, 31))
+        ax1.set_ylim(19.0, 21.5)
+        ax1.set_title(year, pad=0)
+        ax2 = ax1.twinx()
+        y2 = df[glofas_var]
+        ax2.plot(x, y2, '-g')
+        ax2.set_ylim(glofas_xlims)
+        # Glofas detections
+        for detection in get_glofas_detections(y2, thresh):
+            a = detection - GLOFAS_DETECTION_WINDOW_BEHIND
+            b = detection + GLOFAS_DETECTION_WINDOW_AHEAD
+            ax2.plot(x[a:b], y2[a:b], '-y', lw=3)
+            ax2.plot(x[detection], y2[detection], 'x', c='m')   
+        if forecast_var is not None:
+            y3 = df[forecast_var]
+            ax2.plot(x, y3, '-g', alpha=0.5)
+            for detection in get_glofas_detections(y3, thresh):
+                a = detection - GLOFAS_DETECTION_WINDOW_BEHIND
+                b = detection + GLOFAS_DETECTION_WINDOW_AHEAD
+                ax2.plot(x[a:b], y3[a:b], '-c', lw=3, alpha=0.5)
+                ax2.plot(x[detection], y3[detection], '^', c='m')
+        iax += 1
+
+
+plot_years(df_final, rp_dict[5])
+
+```
+
+### Forecasts
```

Comment:
I'm not really following these graphs. What is 'Number'?

Reply:
Ah sorry these figures are indeed unclear, I've now added an explantion